### PR TITLE
fix(typescript-estree): wrap import = declaration in an export node

### LINKED ISSUE #4130: An exported `TSImportEqualsDeclaration` should be wrapped in an `ExportNamedDeclaration` instead of having `.isExport = true`
Code from #4129

```ts
namespace Bar {
  export const bar = 123;
}

export namespace Foo {
  export import TheBar = Bar;
}
```

I didn't know that `export import` was even valid at all in TS.
Right now this will emit a `TSImportEqualsDeclaration` with `.isExported = true`.
This is wrong because it means that rules have to have special handling specifically for this case in order to understand `TheBar` is exported.

We should breaking change fix the AST here so that the AST is instead `ExportNamedDeclaration > TSImportEqualsDeclaration` to align with everything else.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
BREAKING CHANGE:
Removes AST property and changes AST structure

## PR Checklist

- [x] Addresses an existing open issue: fixes #4130
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

A note from our pairing: it's fine to create as many (multiple) nodes as we like from one original node.

Co-authored-by: Brad Zacher <brad.zacher@gmail.com>

### CODE REVIEW

File: packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/entity-name-many/snapshots/1-TSESTree-AST.shot [hunk 1]
```diff
@@ -17,7 +17,6 @@ Program {
         },
       },
       importKind: "value",
-      isExport: false,
```
Comment (JoshuaKGoldberg) [inline packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/entity-name-many/snapshots/1-TSESTree-AST.shot:20]: I have more work to do on fixing up these snapshots. TODO.

File: packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/entity-name-many/snapshots/5-AST-Alignment-AST.shot [hunk 2]
```diff
@@ -2,5 +2,88 @@
...
+          name: 'F',
+
+          range: [7, 8],
+          loc: {
+            start: { column: 7, line: 1 },
+            end: { column: 8, line: 1 },
+          },
+        },
+        importKind: 'value',
++       isExport: false,
```
Comment (JoshuaKGoldberg) [inline packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/entity-name-many/snapshots/5-AST-Alignment-AST.shot:24]: @bradzacher turns out I wasn't done with this - there are still a few `.shot` files with `isExport: false`. `yarn jest spec.ts` in `packages/typescript-estree` shows the failures.

But, on astexplorer.net, `@babel/parser` and other Babel parsers throw an error trying to parse code like `import F = A.B.C;`

Am I missing something about how these are generated?

Comment (JoshuaKGoldberg) [reply packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/entity-name-many/snapshots/5-AST-Alignment-AST.shot:24]: I'll try the changes in https://github.com/typescript-eslint/typescript-eslint/issues/6065 instead.

File: packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts [hunk 3]
```diff
@@ -371,6 +371,16 @@ tester.addFixturePatternConfig('typescript/basics', {
...
     'import-type-error',
+
+    /**
+     * Babel's 'typescript' transform gives these TypeScript-like properties
+     * such as `isExport: false`, but we don't include those in our AST.
+     */
+    'import-equal-declaration',
+    'import-export-equal-declaration',
+    'import-equal-type-declaration',
+    'import-export-equal-type-declaration',
```
Comment (JoshuaKGoldberg) [inline packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts:382]: @bradzacher I'm not 100% that this is accurate, but hey - it got the tests to pass! 🙃

Comment (bradzacher) [reply packages/typescript-estree/tests/ast-alignment/fixtures-to-test.ts:382]: this is one way to do it.
the other way would be do add a transformer to mutate the AST to match babels.

either way sucks.

this is why the ast-spec explicitly snapshots the differences so we don't need to maintain any hacks

**Review by bradzacher — APPROVED** [review]
futureisnow_oldman

**Pull request merged**